wolfcrypt/src/asn.c

```diff
@@ -7921,9 +7921,11 @@ static int EncodeCert(Cert* cert, DerCert* der, RsaKey* rsaKey, ecc_key* eccKey,
     /* SKID */
     if (cert->skidSz) {
         /* check the provided SKID size */
-        if (cert->skidSz > (int)sizeof(der->skid))
+        if (cert->skidSz > (int)sizeof(cert->skid))
```

Comment:
The destination is larger. This check is making sure that skidSz isn't exceeding the cert. Otherwise cert->skidSz could copy more than its max into der->skid.